src/dawn/IIR/MultiStage.cpp

```diff
@@ -423,11 +423,12 @@ json::json MultiStage::jsonDump() const {
   }
   node["Caches"] = cachesJson;
 
-  int cnt = 0;
+  auto stagesArray = json::json::array();
   for(const auto& stage : children_) {
-    node["Stage" + std::to_string(cnt)] = stage->jsonDump(metadata_);
-    cnt++;
+    stagesArray.push_back(stage->jsonDump(metadata_));
   }
+  node["Stages"] = stagesArray;
+
```

Comment:
It does not matter if we have newlines before return stmt's. I would  stick to no (and remove this) to keep the style consistents with what you did in Stage.cpp /Stencil.cpp.

Reply:
done